apps/07jquery-tic-tac-toe/script.js

```diff
@@ -2,4 +2,97 @@
 
 $(document).on('ready', function() {
   // Put app logic in here
+ var playerTurn = 'X';
+ $('[data-cell]').click(function(){
+   $(this).text(playerTurn);
+   checkForWin();
+   playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+
+ });
+  $('button').click(function(){
+    $('[data-cell]').text('');
+    playerTurn = 'X';
+  });
+  function checkForWin(){
+  var cell0 = $('[data-cell="0"]').text();
+  var cell1 = $('[data-cell="1"]').text();
+  var cell2 = $('[data-cell="2"]').text();
+  var cell3 = $('[data-cell="3"]').text();
+  var cell4 = $('[data-cell="4"]').text();
+  var cell5 = $('[data-cell="5"]').text();
+  var cell6 = $('[data-cell="6"]').text();
+  var cell7 = $('[data-cell="7"]').text();
+  var cell8 = $('[data-cell="8"]').text();
+  //top row
+   if(cell0 === playerTurn && cell1 === playerTurn && cell2 === playerTurn){
+     $('#announce-winner').text('Player '+ playerTurn+ ' Wins!');
+   }
+   //middle row
+   if(cell3 === playerTurn && cell4 === playerTurn && cell5 === playerTurn){
+     $('#announce-winner').text('Player '+ playerTurn+ ' Wins!');
+   }
+   //bottom row
+   if(cell6 === playerTurn && cell7 === playerTurn && cell8 === playerTurn){
+     $('#announce-winner').text('Player '+ playerTurn+ ' Wins!');
+   }
+   //left col
+   if(cell0 === playerTurn && cell3 === playerTurn && cell6 === playerTurn){
+     $('#announce-winner').text('Player '+ playerTurn+ ' Wins!');
+   }
+   //middle col
+   if(cell1 === playerTurn && cell4 === playerTurn && cell7 === playerTurn){
+     $('#announce-winner').text('Player '+ playerTurn+ ' Wins!');
+   }
+   //right col
+   if(cell2 === playerTurn && cell5 === playerTurn && cell8 === playerTurn){
+     $('#announce-winner').text('Player '+ playerTurn+ ' Wins!');
+   }
+   //diag top left to bot right
+   if(cell0 === playerTurn && cell4 === playerTurn && cell8 === playerTurn){
+     $('#announce-winner').text('Player '+ playerTurn+ ' Wins!');
+   }
+   //diag top right to bot left
+   if(cell2 === playerTurn && cell4 === playerTurn && cell6 === playerTurn){
+     $('#announce-winner').text('Player '+ playerTurn+ ' Wins!');``
+   }
```

Comment:
You could also use a for loop to short-hand your code, running through each jQuery object and checking for playerTurn.

Example:

```    
var winningPositions = [[0,1,2],[3,4,5][6,7,8]];
for(var i = 0; i < 8; i++){
        if ($('[data-cell="'+ winningPositions[i][0] +'"]').text() === playerTurn && ...
       ....```